### PR TITLE
CORE-627: In Core Crypto, Implement cryptoAmountAdd() for negative arguments

### PR DESCRIPTION
**Pull request by EBGToo**

### CODE REVIEW

File: Swift/BRCryptoTests/BRCryptoAmountTests.swift [hunk 1]
```diff
@@ -116,8 +116,8 @@ class BRCryptoAmountTests: XCTestCase {
         XCTAssert (-1.5       == btc4.double (as: BTC_BTC))
         XCTAssert (-150000000 == btc4.double (as: BTC_SATOSHI))
 
-        XCTAssert ("-B1.50"          == btc4.string (as: BTC_BTC))
-        XCTAssert ("-SAT150,000,000" == btc4.string (as: BTC_SATOSHI))
+        XCTAssertEqual ("-B 1.50", btc4.string (as: BTC_BTC))
```
Comment (michael-brd) [inline Swift/BRCryptoTests/BRCryptoAmountTests.swift]: These tests all fail for me, with the addition of the space. Am I doing something wrong?

Comment (carlcherry) [reply Swift/BRCryptoTests/BRCryptoAmountTests.swift]: I didn't run the unit tests - but the changes to cryptoAmountAdd give me the correct result when calculating a balance.

Comment (EBGToo) [reply Swift/BRCryptoTests/BRCryptoAmountTests.swift]: The 'extra whitespace' appeared out-of-the-blue for me; apparently from the Apple NSNumberFormatter:
```
    private func formatterWith (unit: Unit) -> NumberFormatter {
        let formatter = NumberFormatter()
        formatter.locale = Locale.current
        formatter.numberStyle = .currency
        formatter.currencySymbol = unit.symbol;
        formatter.generatesDecimalNumbers = 0 != unit.decimals
        formatter.maximumFractionDigits = Int(unit.decimals)
        return formatter
    }
```
It is not actually a 'space' but some other whitespace.  It is failing for you in Xcode or in Android Studio?

Comment (michael-brd) [reply Swift/BRCryptoTests/BRCryptoAmountTests.swift]: Checking out this commit, running the tests in XCode, I get a failure. The added space seems to be the culprit.

Comment (michael-brd) [reply Swift/BRCryptoTests/BRCryptoAmountTests.swift]: I think I am using an iOS 12 simulator; perhaps this is tied to a change in iOS 13?

**Pull request merged**